File: project/scratch/standardize_secrets.js
const fs = require('fs');
const path = require('path');

const directory = 'c:\\Users\\Mayibongwe\\Documents\\GitHub\\24-7-DIgiHealth\\project\\app\\api';
const targetSecret = "'secret123!'";
const patterns = [
    /process\.env\.JWT_SECRET \|\| 'secret'/g,
    /process\.env\.JWT_SECRET \|\| 'fallback-secret-for-dev-only'/g
];

const replacement = "process.env.JWT_SECRET || 'secret123!'";

function walk(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walk(fullPath);
        } else if (file.endsWith('.ts') || file.endsWith('.tsx')) {
            let content = fs.readFileSync(fullPath, 'utf8');
            let newContent = content;
            for (const pattern of patterns) {
                newContent = newContent.replace(pattern, replacement);
            }
            if (newContent !== content) {
                fs.writeFileSync(fullPath, newContent);
                console.log(`Updated ${fullPath}`);
            }
        }
    }
}

walk(directory);

File: project/scratch/standardize_secrets.py
import os
import re

directory = r'c:\Users\Mayibongwe\Documents\GitHub\24-7-DIgiHealth\project\app\api'
target_secret = "'secret123!'"
patterns = [
    r"process\.env\.JWT_SECRET \|\| 'secret'",
    r"process\.env\.JWT_SECRET \|\| 'fallback-secret-for-dev-only'"
]

replacement = r"process.env.JWT_SECRET || 'secret123!'"

for root, dirs, files in os.walk(directory):
    for file in files:
        if file.endswith('.ts') or file.endswith('.tsx'):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            new_content = content
            for pattern in patterns:
                new_content = re.sub(pattern, replacement, new_content)
            
            if new_content != content:
                with open(path, 'w', encoding='utf-8') as f:
                    f.write(new_content)
                print(f"Updated {path}")
